### DISCUSSION TITLE
How to get account from browser extension

### DISCUSSION THREAD

**Discussion by o-az** [aptos-labs/aptos-developer-discussions#508] [Questions]
### Discord user ID

amorfati

### Describe your question in detail.

In my own sdk I use the following to access Aptos account through private key:

```ts
const aptosAccount = Account.fromPrivateKey({
  privateKey: ...
})
```

How can i do the same browser side? `window.aptos.account` and `window.aptos.getAccount()` both return account info. I need account of `Account` that I can pass to this:

```ts
const senderAuthenticator = aptos.transaction.sign({
  signer: account // <-- this
})
```

### What have you tried or looked at?  Or how can we reproduce the error?

`window.aptos.account` and `window.aptos.getAccount()` both return account info. I need access to `Account` which I can use to pass to `aptos.transaction.sign`

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

N/A

### Describe your environment or tooling in detail

I'm not using any wallet sdks. I'm using Petra chromium extension and able to connect using `window.aptos.connect()`.

**Comment by 0xmaayan** [comment]
When you retrieve the account from a browser extension, you will _never_ get the account `privateKey` back as it a major security issue - imagine the wallet returns your account private key to any website that asks for it.

Therefore, when you interact with an extension wallet, only the wallet itself knows the connected account private key and can sign transaction on the account behalf. 

If you want the wallet extension to sign / signAndSubmit transaction, you can use the wallet API `.sign()` or `.signAndSubmitTransaction()`. Best would be to work with the [aptos-wallet-adapter](https://github.com/aptos-labs/aptos-wallet-adapter) tool as it handles the different wallet API interfaces and communication.

**Reply by o-az** [reply]
I don't want direct pkey access. I'm coming from working with EVM and Cosmos SDK for years where you can pass the signer down and then construct your transaction. No access to private keys, just the signer. Sudo code:

```
connectWallet()

signer = getSigner()
hash = client.signTx(signer, txPayload)
```

What I ended up doing is simply just passing down `window.aptos` to my SDK

**Reply by Pcharlesme** [reply]
Is this answer or you want me to contribute to that
